cpp/velox/compute/WholeStageResultIterator.cc

```diff
@@ -572,6 +576,12 @@ std::unordered_map<std::string, std::string> WholeStageResultIterator::getQueryC
     configs[velox::core::QueryConfig::kSparkLegacyStatisticalAggregate] =
         std::to_string(veloxCfg_->get<bool>(kSparkLegacyStatisticalAggregate, false));
 
+#ifdef GLUTEN_ENABLE_GPU
+    if (!enableCudf_) {
```

Comment:
@jinchengchenghh

Is there any reason for passing `enableCudf` by method calls down to here, instead of reading a configuration option? I am a bit concerned about such many APIs have to be changed for adding the parameter?